app/assets/javascripts/client/common_create/CreateService.js

```diff
@@ -6,16 +6,35 @@
   CreateService.$inject = [
     '$window',
     '$location',
-    'Assessment'
+    '$stateParams',
+    'Assessment',
+    'Inventory',
+    'Participant',
+    'InventoryParticipant'
   ];
 
-  function CreateService($window, $location, Assessment) {
+  function CreateService($window, $location, $stateParams, Assessment, Inventory, Participant, InventoryParticipant) {
     var service = this;
 
     this.loadDistrict = function(district) {
       this.district = district;
     };
 
+    this.extractCurrentDistrict = function(user, entity) {
+      var result = user.districts[0];
+      for (var i = 0; i < user.districts.length; i++) {
+        if (entity.district_id === user.districts[i].id) {
+          result = user.districts[i];
+          break;
+        }
+      }
+      return result;
+    };
```

Comment:
Underscore can abstract this complexity away for us: `return _.findWhere(user.districts, {id: entity.district_id}) || user.districts[0]`